02-peer-protocol.md

```diff
@@ -445,6 +828,343 @@ to broadcast the commitment transaction to get his funds back and open a new
+return channel-assignable errors before the *accepter*'s revocation
+basepoint is known.
+
+`funding_feerate_perkw` indicates the fee rate that the opening node will
+pay for the funding transaction in satoshi per 1000-weight, as described
+in [BOLT-3, Appendix F](03-transactions.md#appendix-f-dual-funded-transaction-test-vectors).
+
+`locktime` is the locktime for the funding transaction.
+
+The receiving node, if the `locktime` or `feerate_funding_perkw` is considered
+out of an acceptable range, may fail the negotiation. However, it is
+recommended that the *accepter* permits the channel open to proceed
+without their participation in the channel's funding.
+
+Note that `open_channel`'s `channel_reserve_satoshi` has been omitted.
+Instead, the channel reserve is fixed at 1% of the total channel balance
+(`open_channel2`.`funding_satoshis` + `accept_channel2`.`funding_satoshis`)
+rounded down to the nearest whole satoshi or the `dust_limit_satoshis`,
+whichever is greater.
+
+Note that `push_msat` has been omitted.
+
+### The `accept_channel2` Message
+
+This message contains information about a node and indicates its
+acceptance of the new channel.
+
+1. type: 65 (`accept_channel2`)
+2. data:
+    * [`channel_id`:`channel_id`]
+    * [`u64`:`funding_satoshis`]
+    * [`u64`:`dust_limit_satoshis`]
+    * [`u64`:`max_htlc_value_in_flight_msat`]
+    * [`u64`:`htlc_minimum_msat`]
+    * [`u32`:`minimum_depth`]
+    * [`u16`:`to_self_delay`]
+    * [`u16`:`max_accepted_htlcs`]
+    * [`point`:`funding_pubkey`]
+    * [`point`:`revocation_basepoint`]
+    * [`point`:`payment_basepoint`]
+    * [`point`:`delayed_payment_basepoint`]
+    * [`point`:`htlc_basepoint`]
+    * [`point`:`first_per_commitment_point`]
+    * [`accept_tlvs`:`tlvs`]
+
+1. `tlv_stream`: `accept_tlvs`
+2. types:
+   1. type: 1 (`option_upfront_shutdown_script`)
+   2. data:
+       * [`u16`:`shutdown_len`]
+       * [`shutdown_len*byte`:`shutdown_scriptpubkey`]
+
+
+Rationale and Requirements are the same as listed above,
+for [`accept_channel`](#the-accept_channel-message) with the following
+additions.
+
+#### Requirements:
+
+The accepting node:
+    - MAY respond with a `funding_satoshis` value of zero.
+
+#### Rationale
+
+The `funding_satoshis` is the amount of bitcoin in satoshis
+the *accepter* will be contributing to the channel's funding transaction.
+
+Note that `accept_channel`'s `channel_reserve_satoshi` has been omitted.
+Instead, the channel reserve is fixed at 1% of the total channel balance
+(`open_channel2`.`funding_satoshis` + `accept_channel2`.`funding_satoshis`)
+rounded down to the nearest whole satoshi or the `dust_limit_satoshis`,
+whichever is greater.
+
+`channel_id` for the `accept_channel2` MUST be derived using a zero-d out
+basepoint for the peer's revocation basepoint.  This allows the peer to
+return channel-assignable errors before the *accepter*'s revocation
+basepoint is known.
```

Comment:
Is that a bad copy-paste? We received the peer's revocation basepoint in `open_channel2` so we should be able to use the final `channel_id` here, can't we?

Reply:
Originally, I thought the same thing and in fact the v0.10.0 implementation of c-lightning did exactly this but @rustyrussell pointed out that it's exceedingly bad networking practice to have an id rely on data contained within the message, as that builds a dependency between knowing who/where to route the message to and the knowledge of how to unpack the message (tho your router needs to know to pull the channel_id out... so hm).

Reply:
That's an interesting argument! But unless I'm missing something, it doesn't hold at all in this case...because you still have `zerod_channel_id` depend on the `revocation_basepoint` inside the message in both `open_channel2` and `accept_channel2`, don't you?

If I read this correctly, `open_channel2` still uses the opener's `revocation_basepoint` inside its own message to compute its `zerod_channel_id`. And similarly, the accepter uses its own `revocation_basepoint` from its `accept_channel2` message to compute its `zerod_channel_id`. So both messages still conflict with @rustyrussell's rule!

What I'm suggesting is that `accept_channel2` actually incorporate information from `open_channel2` (a different message) to provide the final `channel_id` since it can! Am I missing the point completely or not?

Reply:
And thinking about this more, I believe we *must* use the final `channel_id` (that combines both `revocation_basepoint`s) in `accept_channel2`.

Imagine you're trying to open two distinct channels at the same time and sent two `open_channel2` to a peer (ok, that's a bit contrived, there may be no reason to do that but still).
They respond with two `accept_channel2`, but which one is for which channel?
You can only know if `accept_channel2` contains a reference to *something* in the corresponding `open_channel2`, which is achieved by using a `channel_id` that includes `open_channel2.revocation_basepoint`.

Reply:
Note for future reviewers: my last two comments were based on a misunderstanding. The `zerod_channel_id` sent in `accept_channel2` is the same value that was received in `open_channel2`. Which is perfectly fine and addresses my comments.